pkg/vulnsrc/osv/osv.go

```diff
@@ -0,0 +1,223 @@
+package osv
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"path/filepath"
+	"strings"
+	"time"
+
+	bolt "go.etcd.io/bbolt"
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/trivy-db/pkg/db"
+	"github.com/aquasecurity/trivy-db/pkg/types"
+	"github.com/aquasecurity/trivy-db/pkg/utils"
+	"github.com/aquasecurity/trivy-db/pkg/vulnsrc/vulnerability"
+	vtypes "github.com/aquasecurity/vuln-list-update/osv"
+)
+
+const (
+	osvDir           = "osv"
+	datasourceFormat = "osv-%s"
+	platformFormat   = "Osv Security Advisories %s"
+
+	//ecosystem names
+	Python = "PyPI"
+	Go     = "Go"
+	Rust   = "crates.io"
+)
+
+var defaultEcosystem = []ecosystem{
+	{name: Python, dir: "python", eventType: "ECOSYSTEM", firstVersion: "0"},
+	{name: Go, dir: "go", eventType: "SEMVER", firstVersion: "0"},
+	{name: Rust, dir: "rust", eventType: "SEMVER", firstVersion: "0.0.0-0"},
+}
+
+type ecosystem struct {
+	name         string
+	dir          string
+	eventType    string
+	firstVersion string
+}
+
+type VulnSrc struct {
+	ecosystem ecosystem
+	dbc       db.Operation
+}
+
+func NewVulnSrc(ecosystemName string) VulnSrc {
+	ecosystem := getEcoSystem(ecosystemName)
+	return VulnSrc{
+		ecosystem: ecosystem,
+		dbc:       db.Config{},
+	}
+}
+
+func (vs VulnSrc) Name() string {
+	switch vs.ecosystem.name {
```

Comment:
add `vulnerability.OsvPyPI`, `vulnerability.OsvGo` and `vulnerability.OsvCratesio` as separate property to `defaultEcosystems` map entries above. Get entry and return that property when name is required

Reply:
added new field in `ecosystem`, changed `Name()`